### ISSUE TITLE
[bitnami/keycloak] When setting database vendor to dev-mem or dev-file container still tries to reach postgres

### ISSUE THREAD

**Issue by valerii15298** [bitnami/charts#15927]
### Name and Version

**bitnami/keycloak latest**

### What architecture are you using?

**`amd64`**

### What steps will reproduce the bug?

Create file `docker-compose.yml` with the following content:
```yaml
version: "3.8"
services:
  keycloak:
    image: docker.io/bitnami/keycloak:latest
    environment:
      - KEYCLOAK_DATABASE_VENDOR=dev-mem
```
And in the same directory run command `docker compose up`
You will see the output:
```
Trying to connect to PostgreSQL server postgresql...
app-keycloak-1  | cannot resolve host "postgresql": lookup postgresql on 127.0.0.11:53: read udp 127.0.0.1:56799->127.0.0.11:53: i/o timeout
```

### Are you using any custom parameters or values?

no

### What is the expected behavior?

I expect keycloak to use in memory store and successfully start

### What do you see instead?

I see that keycloak server tries to connect to postgres database while it shouldn't do so according to `KEYCLOAK_DATABASE_VENDOR` env variable

### Additional information

What interesting is that for `docker.io/bitnami/keycloak:21` everything works fine without any bugs

**Comment by javsalgar** [comment]
Hi,

I was unable to reproduce the issue

```
» docker run --name keycloak -e KEYCLOAK_DATABASE_VENDOR=dev-file bitnami/keycloak:latest

Unable to find image 'bitnami/keycloak:latest' locally
latest: Pulling from bitnami/keycloak
5cbf39f0a9ca: Pull complete
Digest: sha256:eb146198a82d2092c10795404218ae5899baa9c7335b6ee290bfaad521a95f0a
Status: Downloaded newer image for bitnami/keycloak:latest
keycloak 08:15:06.66
keycloak 08:15:06.67 Welcome to the Bitnami keycloak container
keycloak 08:15:06.67 Subscribe to project updates by watching https://github.com/bitnami/containers
keycloak 08:15:06.68 Submit issues and feature requests at https://github.com/bitnami/containers/issues
keycloak 08:15:06.68
keycloak 08:15:06.68 INFO  ==> ** Starting keycloak setup **
keycloak 08:15:06.70 INFO  ==> Validating settings in KEYCLOAK_* env vars...
keycloak 08:15:06.74 INFO  ==> Configuring database settings
keycloak 08:15:06.76 INFO  ==> Enabling statistics
keycloak 08:15:06.77 INFO  ==> Configuring http settings
keycloak 08:15:06.80 INFO  ==> Configuring hostname settings
keycloak 08:15:06.81 INFO  ==> Configuring cache count
keycloak 08:15:06.82 INFO  ==> Configuring log level
keycloak 08:15:06.84 INFO  ==> Configuring proxy

keycloak 08:15:06.86 INFO  ==> ** keycloak setup finished! **
keycloak 08:15:06.88 INFO  ==> ** Starting keycloak **
Updating the configuration and installing your custom providers, if any. Please wait.
2023-04-04 08:15:12,426 WARN  [org.keycloak.services] (build-24) KC-SERVICES0047: metrics (org.jboss.aerogear.keycloak.metrics.MetricsEndpointFactory) is implementing the internal SPI realm-restapi-extension. This SPI is internal and may change without notice
2023-04-04 08:15:13,257 WARN  [org.keycloak.services] (build-24) KC-SERVICES0047: metrics-listener (org.jboss.aerogear.keycloak.metrics.MetricsEventListenerFactory) is implementing the internal SPI eventsListener. This SPI is internal and may change without notice
2023-04-04 08:15:25,589 INFO  [io.quarkus.deployment.QuarkusAugmentor] (main) Quarkus augmentation completed in 15810ms
2023-04-04 08:15:29,272 INFO  [org.keycloak.quarkus.runtime.hostname.DefaultHostnameProvider] (main) Hostname settings: Base URL: <unset>, Hostname: <request>, Strict HTTPS: false, Path: <request>, Strict BackChannel: false, Admin URL: <unset>, Admin: <request>, Port: -1, Proxied: true
2023-04-04 08:15:31,986 WARN  [io.quarkus.agroal.runtime.DataSources] (main) Datasource <default> enables XA but transaction recovery is not enabled. Please enable transaction recovery by setting quarkus.transaction-manager.enable-recovery=true, otherwise data may be lost if the application is terminated abruptly
```

Could you try with this tag? [21.0.2-debian-11-r1](https://hub.docker.com/layers/bitnami/keycloak/21.0.2-debian-11-r1/images/sha256-2b75d73742725169d7b12d5518be86f3d1ce61305808703913cd1c170703588d?context=explore)

**Comment by valerii15298** [comment]
@javsalgar Hmm, okay I deleted my docker image `bitnamy/keycloak` with tag `latest` from my docker images(Im using Windows docker setup). And after that, the issue is fixed. But I wonder in the first place why did it happen? If you guys push new version with tag `latest` my laptop should download it, right?
Otherwise, tag `latest` doesn't have any meaning. Let's say I download `bitnami/keycloak:latest` today and the actual version will be 21, then I will be stuck with this version forever using `latest` tag, even when you will push 22, 23 versions.

Could it be my docker local setup cache behavior(though I did not change any default settings)? Or maybe I do not understand fully how docker works.
